src/main/java/com/example/demo/Controllers/MovieCatalogController.java

```diff
@@ -0,0 +1,23 @@
+package com.example.demo.Controllers;
+
+import javax.validation.Valid;
+
+import com.example.demo.Models.MovieCatalog;
+import com.example.demo.Repositories.MovieCatalogRepository;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class MovieCatalogController {
+    private final MovieCatalogRepository movieCatalogRepository;
+
+    public MovieCatalogController(MovieCatalogRepository repository) {
+        this.movieCatalogRepository = repository;
+    }
+
+    @PostMapping("/api/catalog-entry")
```

Comment:
Just out of curiosity, why do you decided to define this path "name" _catalog-entry_?

Reply:
This name was given in the details of the practice as a suggestion, and I couldn't came with a better one

Reply:
Ok, it is fine but it is better to have just a noun as a Resource path, you could use for example just  /catalogs in plural in that way when you will need:
- An endpoint go get all data of catalogs your endpoint should be like:
   GET /catalogs
- An endpoint to get an specific catalog data:
  GET /catalogs/{id}
 - An endpoint to create a new catalog "entry":
  POST /catalogs
  With a Request body that describe your "entry" data
Defining a path sometimes is complex, but you should try to mantain the simplicity in order to avoid long paths and follow the good practices about it.